peps/pep-0744.rst

```diff
@@ -0,0 +1,566 @@
+good compromise between always turning it on and not having it available at all.
+
+Support multiple compiler toolchains
+------------------------------------
+
+Clang is specifically needed because it's the only C compiler with support for
+guaranteed tail calls (|musttail|_), which are required by CPython's
+`continuation-passing-style
+<https://en.wikipedia.org/wiki/Continuation-passing_style#Tail_calls>`_ approach
+to JIT compilation. Without it, the tail-recursive calls between templates could
+result in unbounded C stack growth (and eventual overflow).
+
+.. |musttail| replace:: ``musttail``
+.. _musttail: https://clang.llvm.org/docs/AttributeReference.html#musttail
+
+Since LLVM also includes other functionalities required by the JIT build process
+(namely, utilities for object file parsing and disassembly), and additional
+toolchains introduce additional testing and maintenance burden, it's convenient
+to only support one major version of one toolchain at this time.
+
+Compile the base interpreter's bytecode
+---------------------------------------
+
+Most of the prior art for copy-and-patch uses it as a fast baseline JIT, whereas
+CPython's JIT is using the technique to compile optimized micro-op traces.
+
+In practice, the new JIT currently sits somewhere between the "baseline" and
+"optimizing" compiler tiers of other dynamic language runtimes. This is because
+CPython uses its specializing adaptive interpreter to collect runtime profiling
+information, which is used to detect and optimize "hot" paths through the code.
+This step is carried out using self-modifying code, a technique which is much
+more difficult to implement with a JIT compiler.
+
+While it's *possible* to compile normal bytecode using copy-and-patch (in fact,
+early prototypes predated the micro-op interpreter and did exactly this), it
+just doesn't seem to provide enough optimization potential as the more granular
+micro-op format.
+
+
+Add GPU support
+---------------
+
+The JIT is currently CPU-only. It does not, for example, offload NumPy array
+computations to CUDA GPUs, as JITs like `Numba
+<https://numba.pydata.org/numba-doc/latest/cuda/overview.html>`_ do.
+
+There is already a rich ecosystem of tools for accelerating these sorts of
+specialized tasks, and CPython's JIT is not intended to replace them. Instead,
+it is meant to improve the performance of general-purpose Python code, which is
+less likely to benefit from deeper GPU integration.
+
+Open Issues
+===========
+
+Speed
+-----
+
+Currently, the JIT is `about as fast as the existing specializing interpreter 
+<https://github.com/faster-cpython/benchmarking-public/blob/main/configs.png>`_
+on most platforms. Improving this is obviously a top priority at this point,
+since providing a significant performance gain is the entire motivation for
+having a JIT at all. A number of proposed improvements are already underway, and
+this ongoing work is being tracked in `GH-115802
```

Comment:
The hard-to-remember convention is gh- for issues and GH- for PRs.
```suggestion
this ongoing work is being tracked in `gh-115802
```

Reply:
(I ask because I don't like it and want to leave the PEP the way it is.) 🙂

Reply:
@ezio-melotti Is it documented somewhere?

---

We can see it in titles of backports like https://github.com/python/cpython/pull/117767:

> [3.12] gh-117233: Detect support for several hashes at hashlib build time (GH-117234)

[gh-117233](https://github.com/python/cpython/issues/117233) is the issue, [GH-117234](https://github.com/python/cpython/pull/117234) is the PR being backported.

(I'm fine with leaving the PEP like this.)